### DISCUSSION TITLE
[Feature] Use metadata in storage template

### DISCUSSION THREAD

**Discussion by Spectre5** [immich-app/immich#1950] [Feature Request]
### The feature

The storage template indicates that "Asset's creation timestamp is used for the datetime information".  I have some images on my computer that where edited some after after the image was taken.  For example, an image that was taken June 22, 2019 but was edited (and has a ctime of) May 23, 2020.  When I upload this image, it is stored in .../2020/2020-05-23 instead of .../2020/2019-06-22.  The UI itself uses the metadata and thus shows the correct date and time of the image in the timeline (see example below), but it exists in the "wrong" place in storage.

Can we get an option to use the metadata creation time when uploading instead of ctime?

![image](https://user-images.githubusercontent.com/31610422/222977276-8ac80b23-5cbb-4ac2-b4d4-ed1668c109b4.png)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by Spectre5** [comment]
@alextran1502 do you have any thoughts on this?

**Reply by alextran1502** [reply]
Yes this will be the future expansion of this feature to use some common EXIF info as a template variable

**Reply by Spectre5** [reply]
Great, thanks!

**Comment by bo0tzz** [comment]
@Spectre5 if the UI is showing the correct date, then running the storage migration job should fix your issue.

**Reply by Spectre5** [reply]
I'll check that out, thanks!
